File: unity/Assets/Scripts/Multiplayer/Players/PlayerUnbanRequest.cs
using System;
using UnityEngine;

namespace Michitai.Multiplayer.Players
{
    /// <summary>
    /// Internal request data for unbanning a player from the game in Unity.
    /// </summary>
    [Serializable]
    internal class PlayerUnbanRequest
    {
        /// <summary>
        /// The ID of the player to unban.
        /// </summary>
        public int player_id;

        /// <summary>
        /// Initializes a new PlayerUnbanRequest.
        /// </summary>
        /// <param name="playerId">The ID of the player to unban.</param>
        public PlayerUnbanRequest(int playerId)
        {
            this.player_id = playerId;
        }
    }
}

File: java/src/main/java/com/michitai/multiplayer/players/PlayerUnbanRequest.java
package com.michitai.multiplayer.players;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * Request for unbanning a player.
 */
public class PlayerUnbanRequest {
    @JsonProperty("player_id")
    private int playerId;

    public PlayerUnbanRequest(int playerId) {
        this.playerId = playerId;
    }

    public int getPlayerId() {
        return playerId;
    }

    public void setPlayerId(int playerId) {
        this.playerId = playerId;
    }
}
